src/content/blog/2022-08-05-new-config-system-part-2.md

```diff
@@ -0,0 +1,512 @@
+
+What this means for shareable configs is that you can insert them directly into the array instead of using `extends`, such as:
+
+```js
+import customConfig from "eslint-config-custom";
+
+export default [
+    customConfig,
+    {
+        files: ["**/*.js", "**/*.cjs"],
+        rules: {
+            "semi": "error",
+            "no-unused-vars": "error"
+        }  
+    },
+    {
+        files: ["**/*.js"],
+        rules: {
+            "no-undef": "error",
+            "semi": "warn"
+        }  
+    }
+];
+```
+
+Here, `customConfig` is inserted first in the array so that it becomes the base of configuration for this file. Each of the following config objects builds upon that base to create the final config for a given JavaScript file.
+
+## Reimagined language options
+
+ESLint has always had a strange mix of options that affected how JavaScript was interpreted. There was the top-level `globals` key that modified available global variables, and `ecmaVersion` and `sourceType` as `parserOptions`, not to mention `env` to add more globals. Perhaps the most confusing is that you had to set both `ecmaVersion` and add an environment like `es6` to enable both the syntax you wanted and ensure that the correct global variables would be available.
+
+In flat config, we moved all keys related to JavaScript evaluation into a new top-level key called `languageOptions`.
+
+### Setting `ecmaVersion` in flat config
+
+The biggest change is that we moved `ecmaVersion` out of `parserOptions` and directly into `languageOptions`. This better reflects this key's new behavior, which is to enable both syntax and global variables based on the specified version of ECMAScript. For example:
+
+```js
+export default [
+    {
+        files: ["**/*.js"],
+        languageOptions: {
+            ecmaVersion: 6
+        }
+    }
+];
+```
+
+This config has downgraded `ecmaVersion` to `6`. Doing so ensures that all of the ES6 syntax and all of the ES6 globals are available. (Any custom parsers used will still receive this value of `ecmaVersion`.)
+
+### Setting `sourceType` in flat config
+
+Next, we moved `sourceType` into `languageOptions`. Similar to `ecmaVersion`, this key affects not just how a file is parser, but also how ESLint evaluates its scope structure. We kept the traditional `"module"` for ESM and `"script"` for scripts, and also added `"commonjs"`, which lets ESLint know that it should treat the file as CommonJS (which also enables CommonJS-specific globals). If you are using `ecmaVersion: 3` or `ecmaVersion: 5`, be sure to set `sourceType: script`, like this:
```

Comment:
I couldn’t get this:   Similar to `ecmaVersion`, this key affects not just how a file is parser, but also how ESLint evaluates its scope structure.

Reply:
Oops. Parser should be parsed